geopandas/geodataframe.py

```diff
@@ -1433,17 +1453,48 @@ def __getitem__(self, key):
                 result.__class__ = DataFrame
         return result
 
+    def _persist_old_default_geometry_colname(self):
+        """Internal util to temporarily persist the default geometry column
+        name of 'geometry' for backwards compatibility."""
+        # self.columns check required to avoid this warning in __init__
+        if (
+            self._geometry_column_name == DEFAULT_GEO_COLUMN_NAME
+            and "geometry" not in self.columns
+        ):
+            msg = (
+                "You are adding a column named 'geometry' to a GeoDataFrame "
+                "constructed without an active geometry column. Currently, "
+                "this automatically sets the active geometry column to 'geometry' "
+                "but in the future that will no longer happen. Instead, either "
+                "provide geometry to the GeoDataFrame constructor "
+                "(GeoDataFrame(... geometry=GeoSeries()) or use `set_geometry` "
```

Comment:
Should we more explicitly say that you can do `.set_geometry("geometry")`? (because this warning is only triggered for that name, right?)

Reply:
correct, makes sense